cxx/src/lib/io/fileio.cc

```diff
@@ -129,4 +210,34 @@ size_t fread_from_file(TimeSeries& d,const string dir, const string dfile,
 		return ns_read;
 	}catch(...){throw;};
 }
+size_t fread_from_file(Ensemble<Seismogram> &de, vector<string> dirs, vector<string> dfiles, 
```

Comment:
This algorithm has some issues.   A reader has some complexity not shared by a writer.   In fact, there are a some design decisions the group needs to discuss:
1.   A problem in ensemble reader is matching Metadata stored in MongoDB with the sample data stored as a block by the writer.   With the current model having TimeSeries and Seismogram atomic each document in a wf colletion defines one TimeSeries or Seismogram with dir/dfile and foff defing where fread should be positioned to load the sample data.   If we stick with that model the algorithm needs to sort documents properly.   The sort requires 2 keys:  dir and dfile.   The reader then could remain stateless by grouping the data by unique dir:dfile keys.  We open and close the file then only when the dir or dfile change.   
2. The alternative is to allow a new structure for ensembles in a wf collection using subdocuments.   That is, the ensemble's entry in a wf collection would have the ensemble metadata at the top level of the document and the member metadata in a subdocument with some unique key to define the subdocument lile "member_metadata

I think the sort - group model is the better solution.  It is more generic and doesn't potentially break other things.